### PR TITLE
Backport 15533

### PR DESCRIPTION
**Pull request by markstory**
Backport the changes from #15533 to 3.x

### CODE REVIEW

**Comment by Rytiss** [comment]
Lacks the critical part of the original change that runs `_encode()` on the consumer key: https://github.com/cakephp/cakephp/pull/15533/files#diff-14e47e708381dc7e32406d133527a97f94d5cad603de3dcca07e7236a2335affR144

**Comment by markstory** [comment]
Interesting I wonder why that chunk didn't get applied when the commit was picked across.

**Comment by ADmad** [comment]
And the test case is added twice :)

**Comment by markstory** [comment]
Should be better now :blush:

File: tests/TestCase/Http/Client/Auth/OauthTest.php [hunk 1]
```diff
@@ -307,6 +307,37 @@ public function testHmacSigning()
...
+            'token' => 'nnch734d00sl2jdk',
+            'nonce' => 'kllo9940pd9333jh',
+            'timestamp' => '1191242096',
+        ];
+        $auth = new Oauth();
+        $request = $auth->authentication($request, $options);
+
+        $result = $request->getHeaderLine('Authorization');
+        $expected = '2hr/eoFyTSuWc6SfZIvkhpeRHdM=';
+        $this->assertStringContainsString(
```
Comment (markstory) [inline tests/TestCase/Http/Client/Auth/OauthTest.php:335]:
```suggestion
        $this->assertContains(
```

**Pull request merged**